src/Fleck/Interfaces/ISocket.cs

```diff
@@ -14,6 +15,7 @@ public interface ISocket
         int RemotePort { get; }
         Stream Stream { get; }
         bool NoDelay { get; set; }
+        Socket Socket { get; }

```

Comment:
I'd rather not have the ISocket interface expose an underlying Socket. It ruins the abstraction. If you need a property of the Socket, just expose that.